libclambcc/Common/ClamBCUtilities.cpp

```diff
@@ -146,3 +146,93 @@ llvm::BasicBlock *getEntryBlock(llvm::BasicBlock *pBlock)
 {
     return llvm::cast<llvm::BasicBlock>(pBlock->getParent()->begin());
 }
+
+int64_t getTypeSize(llvm::Module * pMod, llvm::Type *pt)
+{
+
+    int64_t size = pt->getScalarSizeInBits();
+    if (size) {
+        return size;
+    }
+
+    if (ArrayType *pat = llvm::dyn_cast<ArrayType>(pt)) {
+        size = pat->getNumElements() * (getTypeSize(pMod, pat->getElementType()));
+        if (size) {
+            return size;
+        }
+    }
+
+    if (StructType *pst = llvm::dyn_cast<StructType>(pt)) {
```

Comment:
Not a fan of initializing in an if statement and then evaluating if not NULL.  This `=` is easy to misinterpret as `==`.  Can you please move initialization out of the `if()` or else make make the `==` explicit so it's easier to read? #